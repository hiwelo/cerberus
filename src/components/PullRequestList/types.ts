import { PullRequestConnection } from '../../types/GitHubApi/PullRequest';

export interface PullRequestListProps {
  /** List of deployments to display */
  list: PullRequestConnection;
}
