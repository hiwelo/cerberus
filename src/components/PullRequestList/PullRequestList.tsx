import React, { FunctionComponent } from 'react';
import { PullRequestCard } from 'components';
import { List, ListItem } from './components';
import { PullRequestListProps } from './types';

export const PullRequestList: FunctionComponent<PullRequestListProps> = ({
  list,
}) => {
  const pullRequestEdges = list.edges;
  const pullRequestList = pullRequestEdges.map(edge => edge.node);

  return (
    <List>
      {pullRequestList.map(pullRequest => (
        <ListItem key={pullRequest.number}>
          <PullRequestCard item={pullRequest} />
        </ListItem>
      ))}
    </List>
  );
};
