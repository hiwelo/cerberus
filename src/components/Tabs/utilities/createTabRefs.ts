import { useRef } from 'react';
import { TabsProps } from '../types';

/**
 * Creates a series of React Ref for each panel part of the Tabs defined for
 * this `Tabs` component
 *
 * @param tabs The dictionary of Tabs for the related Tabs component
 */
export function createTabRefs(tabs: TabsProps['tabs']): TabsProps['tabs'] {
  // Transforms the TabListDictionary as an array of tabs
  const tabsEntries = Object.entries(tabs);
  const tabsWithRefs: TabsProps['tabs'] = {};

  // Loops over each tab and creates a React Ref for each panel HTMLElement
  tabsEntries.forEach(([tabKey, tabInfo]) => {
    const refPanel = useRef<HTMLDivElement>(null);

    tabsWithRefs[tabKey] = { ...tabInfo, refPanel };
  });

  return tabsWithRefs;
}
