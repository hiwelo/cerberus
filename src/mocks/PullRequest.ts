import {
  PullRequest,
  PullRequestConnection,
} from 'types/GitHubApi/PullRequest';
import { mockedActor, mockedCommit, mockedDeployment } from '.';

export const mockedPullRequest: PullRequest = {
  author: mockedActor,
  closed: false,
  locked: false,
  merged: false,
  number: 123,
  permalink: 'https://github.com/owner/repository/pull/123',
  title: 'Test of pull request',
  url: 'https://github.com/owner/repository/pull/123',
  commits: {
    edges: [
      {
        __typename: 'CommitEdge',
        node: {
          commit: {
            ...mockedCommit,
            deployments: {
              edges: [
                {
                  __typename: 'DeploymentEdge',
                  node: mockedDeployment,
                },
              ],
            },
          },
        },
      },
    ],
  },
};

export const mockedPullRequestConnection: PullRequestConnection = {
  edges: [
    {
      __typename: 'PullRequestEdge',
      node: mockedPullRequest,
    },
  ],
};
