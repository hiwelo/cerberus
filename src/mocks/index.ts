export { mockedActor } from './Actor';
export { mockedCommit } from './Commit';
export { mockedDeployment } from './Deployment';
export { mockedPullRequest } from './PullRequest';
