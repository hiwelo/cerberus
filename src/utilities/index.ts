export { breakpoints } from './breakpoints';
export { listNodesFromConnection } from './listNodesFromConnection';
