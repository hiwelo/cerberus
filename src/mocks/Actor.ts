import { Actor } from 'types/GitHubApi/Actor';

/**
 * Mocked Actor object
 */
export const mockedActor: Actor = {
  avatarUrl: 'https://i.pravatar.cc/300',
  login: 'test',
  url: 'https://github.com/test',
};
