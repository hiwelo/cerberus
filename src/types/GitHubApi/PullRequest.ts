import { Actor } from './Actor';
import { CommitConnection } from './Commit';
import { URI } from './GitHubApi';

/** A repository pull request */
export interface PullRequest {
  /** The actor who authored the comment */
  author: Actor;
  /** true if the pull request is closed */
  closed: boolean;
  /** true if the pull request is locked */
  locked: boolean;
  /** Whether or not the pull request was merged */
  merged: boolean;
  /** Identifies the pull request number */
  number: number;
  /** The permalink to the pull request */
  permalink: URI;
  /** Identifies the pull request title */
  title: string;
  /** The HTTP URL for this pull request */
  url: URI;
  /** The body rendered to text */
  bodyText?: string;
  /** A list of commits present in this pull request's head branch not present in the base branch */
  commits?: CommitConnection;
}

/** The connection type for PullRequest */
export interface PullRequestConnection {
  edges: PullRequestEdge[];
}

/** A pull request node in a connection */
export interface PullRequestEdge {
  __typename: 'PullRequestEdge';
  node: PullRequest;
}
