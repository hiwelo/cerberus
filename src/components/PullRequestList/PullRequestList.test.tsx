import React from 'react';
import { mockedPullRequestConnection } from 'mocks/PullRequest';
import { testA11y, testSnapshot } from 'utilities/test';
import { PullRequestList } from './PullRequestList';

const TestContainer = (
  <PullRequestList list={mockedPullRequestConnection}>Test</PullRequestList>
);

describe('PullRequestList', () => {
  testSnapshot(TestContainer);
  testA11y(TestContainer);
});
