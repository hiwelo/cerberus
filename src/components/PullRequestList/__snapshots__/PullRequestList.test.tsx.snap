// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequestList should match snapshot 1`] = `
.c7 {
  position: relative;
  color: #006e97;
  font-weight: 500;
}

.c7:focus {
  outline: none;
}

.c11 {
  display: inline-block;
  padding: 0.8rem 1.6rem 0.7rem;
  background-color: transparent;
  border: 0.3rem solid #006e97;
  border-radius: 0.8rem;
  color: #222222;
  font-weight: 500;
  line-height: 1.2em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: background-color 0.3s ease-in-out,color 0.2s ease-in-out;
  transition: background-color 0.3s ease-in-out,color 0.2s ease-in-out;
}

.c11:focus,
.c11:hover {
  background-color: #006e97;
  color: #fafaff;
}

.c10 {
  margin-bottom: 1.6rem;
}

.c10:not(:last-of-type) {
  margin-right: 1.6rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0;
  list-style: none;
}

.c8 {
  position: relative;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8::before {
  position: absolute;
  bottom: 0.1em;
  left: 0;
  right: 0;
  content: '';
  height: 0.15em;
  background-color: #e8c000;
  border-radius: 0.4rem;
}

.c8:focus {
  outline: none;
}

.c8:focus::before,
.c8:hover::before {
  background-color: #8ab0ab;
  bottom: 0;
}

.c2 {
  margin-bottom: 0.8rem;
}

.c5 {
  margin-right: 0.4em;
  font-weight: 300;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 1.6rem;
  font-size: 1.4rem;
}

.c6 {
  margin: 0;
  margin-right: 1.6rem;
  font-weight: 400;
}

.c3 {
  display: inline-block;
  margin: 0;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-letter-spacing: -0.005em;
  -moz-letter-spacing: -0.005em;
  -ms-letter-spacing: -0.005em;
  letter-spacing: -0.005em;
}

.c0 {
  list-style: none;
  margin: 2.4rem auto 0;
  padding: 0;
}

.c1 {
  position: relative;
  margin: 0 1.6rem 3.2rem;
  padding: 0 0 1.6rem;
}

.c1:last-of-type {
  margin-bottom: 0;
}

.c1:not(:last-of-type)::after {
  position: absolute;
  left: 0;
  right: 0;
  top: calc(100% - 0.2rem);
  content: '';
  display: block;
  height: 0.4rem;
  background: #8ab0ab;
  border-radius: 0.2rem;
}

@media (prefers-reduced-motion:no-preference) {
  .c8::before {
    -webkit-transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
    transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
  }
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <article
      class=""
    >
      <header
        class="c2"
      >
        <h3
          class="c3"
        >
          Test of pull request
        </h3>
      </header>
      <dl
        class="c4"
      >
        <dt
          class="c5"
        >
          PR
        </dt>
        <dd
          class="c6"
        >
          <a
            class="c7 c8"
            href="https://github.com/owner/repository/pull/123"
            rel="noopener noreferrer"
            target="_blank"
          >
            #123
          </a>
        </dd>
        <dt
          class="c5"
        >
          Author
        </dt>
        <dd
          class="c6"
        >
          <a
            class="c7 c8"
            href="https://github.com/test"
            rel="noopener noreferrer"
            target="_blank"
          >
            test
          </a>
        </dd>
      </dl>
      <div
        class=""
      >
        <ul
          class="c9"
        >
          <li
            class="c10"
          >
            <a
              class="c7 c11"
              href="https://example.com/"
            >
              test
            </a>
          </li>
        </ul>
      </div>
    </article>
  </li>
</ul>
`;
